src-care/topkg_care_delegate.ml

```diff
@@ -80,7 +80,10 @@ let publish_in_git_branch ~branch ~name ~version ~docdir ~dir =
     let msg = strf "Update %s doc to %s." name version in
     Topkg.Vcs.get ()
     >>= fun repo -> Ok (git_for_repo repo)
-    >>= fun git -> OS.Cmd.run Cmd.(git % "checkout" % branch)
+    >>= fun git -> OS.Cmd.run Cmd.(git % "remote" % "add" % "upstream" % remote)
+    >>= fun () -> OS.Cmd.run Cmd.(git % "checkout" % "-b" % branch)
+    >>= fun () -> OS.Cmd.run Cmd.(git % "fetch" % "upstream")
+    >>= fun () -> OS.Cmd.run Cmd.(git % "reset" % ("upstream/"^branch) % "--hard")
     >>= fun () -> delete dir
```

Comment:
There are a little bit to much ad-hoc naming in there to my taste. We should go back to what we had before with a single `remote` symbolic argument for the library. More ad-hoc dancing can be done in the delegate.